ui/src/dashboards/components/createFromTemplateOverlay/DashboardCreateFromTemplateOverlay.tsx

```diff
@@ -0,0 +1,183 @@
+            <Overlay.Heading
+              title="Create Dashboard from a Template"
+              onDismiss={this.onDismiss}
+            />
+            <Overlay.Body>{this.overlayBody}</Overlay.Body>
+            <Overlay.Footer>
+              <Button
+                text="Cancel"
+                onClick={this.onDismiss}
+                key="cancel-button"
+              />
+              <Button
+                text="Create Dashboard"
+                onClick={this.onSubmit}
+                key="submit-button"
+                testID="create-dashboard-button"
+                color={ComponentColor.Success}
+                status={this.submitStatus}
+              />
+            </Overlay.Footer>
+          </Overlay.Container>
+        </Overlay>
+      </GetResources>
+    )
+  }
+
+  private get overlayBody(): JSX.Element {
+    const {
+      selectedTemplateSummary,
+      cells,
+      variables,
+      selectedTemplate,
+    } = this.state
+    const {templates} = this.props
+
+    if (!templates.length) {
+      return <DashboardTemplatesEmpty />
+    }
+
+    return (
+      <DashboardTemplateBrowser
+        templates={templates}
+        cells={cells}
+        variables={variables}
+        selectedTemplate={selectedTemplate}
+        selectedTemplateSummary={selectedTemplateSummary}
+        onSelectTemplate={this.handleSelectTemplate}
+      />
+    )
+  }
+
+  private get submitStatus(): ComponentStatus {
+    const {selectedTemplate} = this.state
+
+    return selectedTemplate ? ComponentStatus.Default : ComponentStatus.Disabled
+  }
+
+  private getVariablesForTemplate(template: ITemplate): string[] {
+    const variables = []
+    const included = template.content.included as IDashboardTemplateIncluded[]
+    included.forEach(data => {
+      if (data.type === TemplateType.Variable) {
+        variables.push(data.attributes.name)
+      }
+    })
+
+    return variables
+  }
+
+  private getCellsForTemplate(template: ITemplate): string[] {
+    const cells = []
+    const included = template.content.included as IDashboardTemplateIncluded[]
+    included.forEach(data => {
+      if (data.type === TemplateType.View) {
+        cells.push(data.attributes.name)
+      }
+    })
+
+    return cells
+  }
+
+  private handleSelectTemplate = async (
+    selectedTemplateSummary: TemplateSummary
+  ): Promise<void> => {
+    const selectedTemplate = await getTemplateByID(selectedTemplateSummary.id)
+    this.setState({
+      selectedTemplateSummary,
+      selectedTemplate,
+      variables: this.getVariablesForTemplate(selectedTemplate),
+      cells: this.getCellsForTemplate(selectedTemplate),
+    })
+  }
+
+  private onDismiss = () => {
+    const {router} = this.props
+    router.goBack()
+  }
+
+  private onSubmit = async (): Promise<void> => {
+    const {createDashboardFromTemplate} = this.props
+
+    await createDashboardFromTemplate(this.state
+      .selectedTemplate as DashboardTemplate)
```

Comment:
This would be much easier to read if they were two steps like:
```
dashboardTemplate = this.state.selectedTemplate as DashboardTemplate
await createDashboardFromTemplate(dashboardTemplate)
```